lib/services/local/okta_test.go

```diff
@@ -0,0 +1,348 @@
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend/memory"
+)
+
+// TestOktaImportRuleCRUD tests backend operations with Okta import rule resources.
+func TestOktaImportRuleCRUD(t *testing.T) {
+	ctx := context.Background()
+	clock := clockwork.NewFakeClock()
+
+	backend, err := memory.New(memory.Config{
+		Context: ctx,
+		Clock:   clock,
+	})
+	require.NoError(t, err)
+
+	service, err := NewOktaService(backend)
+	require.NoError(t, err)
+
+	// Create a couple Okta import rule.
+	importRule1, err := types.NewOktaImportRule(
+		types.Metadata{
+			Name: "importRule1",
+		},
+		types.OktaImportRuleSpecV1{
+			Mappings: []*types.OktaImportRuleMappingV1{
+				{
+					Match: []*types.OktaImportRuleMatchV1{
+						{
+							AppIDs: []string{"yes"},
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+				{
+					Match: []*types.OktaImportRuleMatchV1{
+						{
+							GroupIDs: []string{"yes"},
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+	importRule2, err := types.NewOktaImportRule(
+		types.Metadata{
+			Name: "importRule2",
+		},
+		types.OktaImportRuleSpecV1{
+			Mappings: []*types.OktaImportRuleMappingV1{
+				{
+					Match: []*types.OktaImportRuleMatchV1{
+						{
+							AppIDs: []string{"yes"},
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+				{
+					Match: []*types.OktaImportRuleMatchV1{
+						{
+							GroupIDs: []string{"yes"},
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	// Initially we expect no import rule.
+	out, nextToken, err := service.ListOktaImportRules(ctx, 200, "")
+	require.NoError(t, err)
+	require.Empty(t, nextToken)
+	require.Empty(t, out)
+
+	// Create both import rules.
+	err = service.CreateOktaImportRule(ctx, importRule1)
+	require.NoError(t, err)
+	err = service.CreateOktaImportRule(ctx, importRule2)
+	require.NoError(t, err)
+
+	// Fetch all import rules.
+	out, nextToken, err = service.ListOktaImportRules(ctx, 200, "")
+	require.NoError(t, err)
+	require.Empty(t, nextToken)
+	require.Empty(t, cmp.Diff([]types.OktaImportRule{importRule1, importRule2}, out,
+		cmpopts.IgnoreFields(types.Metadata{}, "ID"),
+	))
+
+	// Fetch a paginated list of import rules
+	paginatedOut := make([]types.OktaImportRule, 0, 2)
+	numPages := 0
+	for {
+		numPages++
+		out, nextToken, err = service.ListOktaImportRules(ctx, 1, nextToken)
+		require.NoError(t, err)
+
+		paginatedOut = append(paginatedOut, out...)
+		if nextToken == "" {
+			break
+		}
+	}
+
+	require.Equal(t, 2, numPages)
```

Comment:
Would it make more sense to get rid of `numPages` and just `require.Len(t, 2, paginatedOut)` instead?